traffic_portal/app/src/common/modules/notifications/notifications.tpl.html

```diff
@@ -0,0 +1,24 @@
+<!--
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+<div id="notificationsContainer">
+    <div ng-show="cdn.notificationCreatedBy" class="alert alert-info" ng-repeat="cdn in cdns">
+        <div ng-bind-html="cdn.name + ': ' + cdn.notification + ' (' + cdn.notificationCreatedBy + ')' | linky:'_blank'"></div>
```

Comment:
What is the `linky` pipe doing here?

Reply:
converts urls to links. i.e. you can have a notification like this:

```
Please go to http://www.website.com for more information
```

and it will turn that into a link.

Reply:
Is that safe? I think using invalidated user input to provide functionality like this could technically be an XSS vulnerability (is "XSS" appropriate here? Not sure)

Reply:
not sure. i could remove it i guess but seemed like nice functionality. i'll see if the library does any sanitation.

Reply:
https://docs.angularjs.org/api/ngSanitize/filter/linky